SNAP-2602 : On snappy UI, add column named "Overflown Size"/ "Disk Size" in Tables.

snappy-sachin opened the pull request:
## Changes proposed in this pull request

 - SnappyRegionStats is updated for holding tables size on disk i.e. tables overflown size.
 - SnappyRegionStatsCollectorFunction is updated to set tables size on disk.

## Patch testing

 - Tested Manually.

## Is precheckin with -Pstore clean?

## ReleaseNotes changes

 - NA

## Other PRs 

SnappyData: https://github.com/SnappyDataInc/snappydata/pull/1185
Spark: https://github.com/SnappyDataInc/spark/pull/127

snappy-sachin committed:
Changes for SNAP-2602:
 - SnappyRegionStats is updated for holding tables size on disk i.e. tables overflown size.
 - SnappyRegionStatsCollectorFunction is updated to set tables size on disk.

snappy-sachin committed:
Merge branch 'snappy/master' into SNAP-2602

sumwale reviewed (commented):
Looks fine but since overflow size is always total-memory, won't it be easier to just calculate this in the end results in front-end? Are you expecting that this calculation could become more involved later?

snappy-sachin commented:
I also had thought to keep calculation of overflown size on front end but could not find it convenient, as I am using JSON response object as an input to tables grid. I tried to avoid function execution for each table cell otherwise it may cause slower re/loading data into table.

sumwale commented:
@snappy-sachin Not in display, but couldn't you calculate it when creating TableSummary (in TableDetails.scala)?

snappy-sachin commented:
Yes, That can be done.
I thought putting this value in SnappyRegionStats itself will help not only web services but also all those  who need Snappy Region Stats.
Having said that, as you said I can keep calculation of on disk size in Table Details to use it in only in web services.

sumwale commented:
Ok, sounds fine. Perhaps in future we may calculate them differently.

snappy-sachin committed:
Merge branch 'snappy/master' into SNAP-2602

snappy-sachin committed:
Code changes for SNAP-2602:
  - Renaming variable name SnappyRegionStats.sizeOnDisk to SnappyRegionStats.sizeSpillToDisk and
    relevant code changes.

snappy-sachin commented:
All feedback comments are taken care. Hence Merging.

sumwale commented:
@snappy-sachin this broke AnalyzeSerializablesJUnitTest. No merges to store should ever be done without "precheckin -Pstore" task being clean (or with only known failures).

snappy-sachin commented:
> @snappy-sachin this broke AnalyzeSerializablesJUnitTest. No merges to store should ever be done without "precheckin -Pstore" task being clean (or with only known failures).

Oops! I missed it. Thank you for adding the fixes. Henceforth, I will take care of such things too.
